sympy/functions/special/gamma_functions.py

```diff
@@ -1015,3 +1005,90 @@ def trigamma(x):
     .. [3] http://functions.wolfram.com/GammaBetaErf/PolyGamma2/
     """
     return polygamma(1, x)
+
+###############################################################################
+##################### COMPLETE MULTIVARIATE GAMMA FUNCTION ####################
+###############################################################################
+
+class multivariate_gamma(Function):
+    r"""
+    The multivariate gamma function is a generalization of the gamma function i.e,
+
+    .. math::
+        \Gamma_p(z) = \pi^{p(p-1)/4}\prod_{k=1}^p \Gamma[z + (1 - k)/2].
+
+    Special case, multivariate_gamma(x, 1) = gamma(x)
+
+    Parameters
+    ==========
+
+    p: order or dimension of the multivariate gamma function
+
+    Examples
+    ========
+
+    >>> from sympy import S, I, pi, oo, gamma, multivariate_gamma
+    >>> from sympy import Symbol
+    >>> x = Symbol('x')
+    >>> p = Symbol('p', positive=True, integer=True)
+
+    >>> multivariate_gamma(x, p)
+    pi**(p*(p - 1)/4)*Product(gamma(-_k/2 + x + 1/2), (_k, 1, p))
+
+    Several special values are known:
+    >>> multivariate_gamma(1, 1)
+    1
+    >>> multivariate_gamma(4, 1)
+    6
+    >>> multivariate_gamma(S(3)/2, 1)
+    sqrt(pi)/2
+
+    Writing multivariate_gamma in terms of gamma function
+    >>> multivariate_gamma(x, 1)
+    gamma(x)
+
+    >>> multivariate_gamma(x, 2)
+    sqrt(pi)*gamma(x)*gamma(x - 1/2)
+
+    >>> multivariate_gamma(x, 3)
+    pi**(3/2)*gamma(x)*gamma(x - 1)*gamma(x - 1/2)
+
+    See Also
+    ========
+
+    gamma, lowergamma, uppergamma, polygamma, loggamma, digamma, trigamma
+    sympy.functions.special.beta_functions.beta
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multivariate_gamma_function
+    """
+    unbranched = True
+
+    def fdiff(self, argindex=2):
+        from sympy import Sum
+        if argindex == 2:
+            x, p = self.args
+            k = Dummy('k')
+            return self.func(x, p)*Sum(polygamma(0, x + (1 - k)/2), (k, 1, p))
+        raise ArgumentIndexError(self, argindex)
+
+    @classmethod
+    def eval(cls, x, p):
+        from sympy import Product
+        x, p = map(sympify, (x, p))
+        if p.is_positive == False or p.is_integer == False:
+            raise ValueError('Order parameter p must be positive integer.')
+        k = Dummy('k', real=True)
+        return (pi**(p*(p - 1)/4)*Product(gamma(x + (1 - k)/2),
+                    (k, 1, p))).doit()
+
+    def _eval_conjugate(self):
+        x, p = self.args
+        return self.func(x.conjugate(), p.conjugate())
+
+    def _eval_is_real(self):
+        x, p = self.args
+        if (x > (p - 1)/2) or x.is_noninteger:
```

Comment:
This doesn't seem right.
```py
>>> expr = multivariate_gamma(S.Half, 3, evaluate=False)
>>> expr.is_real == expr.doit().is_real
False
```

Reply:
To be precise, this was incorrect not because one of them was `None`, but due to one being `True` and the other being `False`:
```py
>>> expr.is_real
True
>>> expr.doit().is_real
False
```